### PR TITLE
langchain_google_vertexai: Add support for SystemMessage for Gemini chat model

### PR DESCRIPTION
**Pull request by RajeshThallam**
- **Description:** In Google Vertex AI, Gemini Chat models currently doesn't have a support for SystemMessage. This PR adds support for it only if a user provides additional convert_system_message_to_human flag during model initialization (in this case, SystemMessage would be prepended to the first HumanMessage). **NOTE:** The implementation is similar to #14824 

- **Twitter handle:** rajesh_thallam

### CODE REVIEW

File: libs/partners/google-vertexai/tests/integration_tests/test_chat_models.py [hunk 1]
```diff
@@ -174,3 +174,39 @@ def test_vertexai_single_call_fails_no_message() -> None:
         str(exc_info.value)
         == "You should provide at least one message to start the chat!"
     )
+
+
+@pytest.mark.parametrize("model_name", model_names_to_test)
+def test_chat_google_genai_system_message_error(model_name: str) -> None:
```
Comment (lkuligin) [inline libs/partners/google-vertexai/tests/integration_tests/test_chat_models.py:180]: nits: it looks like copy-paste artifacts, could you rename the test, please?

**Comment by lkuligin** [comment]
SGTM

@efriis could you take a look, please?

File: libs/partners/google-vertexai/langchain_google_vertexai/chat_models.py [hunk 2]
```diff
@@ -141,8 +143,23 @@ def _convert_to_prompt(part: Union[str, Dict]) -> Part:
...
+                """SystemMessages are not yet supported!
+                
+To automatically convert the leading SystemMessage to a HumanMessage,
+set  `convert_system_message_to_human` to True. Example:
+
+llm = ChatVertexAI(model_name="gemini-pro", convert_system_message_to_human=True)
+"""
+            )
+        elif i == 0 and isinstance(message, SystemMessage):
+            raw_system_message = message
```
Comment (efriis) [inline libs/partners/google-vertexai/langchain_google_vertexai/chat_models.py:161]: I think we might need a `raw_system_message = None` above

File: libs/partners/google-vertexai/langchain_google_vertexai/chat_models.py [hunk 3]
```diff
@@ -156,6 +173,16 @@ def _convert_to_prompt(part: Union[str, Dict]) -> Part:
         if isinstance(raw_content, str):
             raw_content = [raw_content]
         parts = [_convert_to_prompt(part) for part in raw_content]
+
+        if raw_system_message:
```
Comment (efriis) [inline libs/partners/google-vertexai/langchain_google_vertexai/chat_models.py:177]: given this logic only applies to the beginning of the loop, it'd be cleaner to handle this before the loop! Any chance you could change that for readability?

File: libs/partners/google-vertexai/langchain_google_vertexai/chat_models.py (hunk 2 above)
Comment (efriis) [reply libs/partners/google-vertexai/langchain_google_vertexai/chat_models.py:161]: (ignore if pulling out of the loop)

File: libs/partners/google-vertexai/tests/integration_tests/test_chat_models.py (hunk 1 above)
Comment (efriis) [reply libs/partners/google-vertexai/tests/integration_tests/test_chat_models.py:180]: +1!

**Review by efriis — COMMENTED** [review]
Thanks! Just some small changes

**Comment by thomasmenard-oa** [comment]
@lkuligin and @RajeshThallam will you be able to edit the code and do the PR this week?
We need to test and integrate it in our sprint before the 25th of jan.

Big thanks !

File: libs/partners/google-vertexai/tests/integration_tests/test_chat_models.py (hunk 1 above)
Comment (RajeshThallam) [reply libs/partners/google-vertexai/tests/integration_tests/test_chat_models.py]: Updated test names

File: libs/partners/google-vertexai/langchain_google_vertexai/chat_models.py (hunk 2 above)
Comment (RajeshThallam) [reply libs/partners/google-vertexai/langchain_google_vertexai/chat_models.py:162]: Added `raw_system_message = None`

File: libs/partners/google-vertexai/langchain_google_vertexai/chat_models.py (hunk 3 above)
Comment (RajeshThallam) [reply libs/partners/google-vertexai/langchain_google_vertexai/chat_models.py:178]: There is a bit more dependency and would make the code redundant if the logic is pulled out of the loop. Also this is a conditional logic that gets applied to element=1 when element=0 meets a condition.

**Comment by RajeshThallam** [comment]
@efriis I have addressed the comments. PTAL.

**Comment by lkuligin** [comment]
SGTM

**Comment by lkuligin** [comment]
@RajeshThallam please, rebase on master

**Comment by RajeshThallam** [comment]
rebased and resolved conflicts

**Review by efriis — APPROVED** [review]

**Review by lkuligin — APPROVED** [review]

**Pull request merged**